apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/SignedUrlUtilsTest.cs

```diff
@@ -0,0 +1,487 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Json;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Xunit;
+
+namespace Google.Storage.V1.IntegrationTests
+{
+    using static TestHelpers;
+
+    [Collection(nameof(StorageFixture))]
+    public class SignedUrlUtilsTest
+    {
+        private readonly string _credentialsFilePath = GetCredentialsFilePath();
+        private readonly StorageFixture _fixture;
+        private readonly HttpClient _httpClient = new HttpClient();
+
+        public SignedUrlUtilsTest(StorageFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async void DeleteTest()
+        {
+            var bucket = _fixture.SingleVersionBucket;
+            var name = GenerateName();
+            var duration = TimeSpan.FromSeconds(5);
```

Comment:
It would be nice to somehow be able to do the first part of all of these tests, then wait 5 seconds, then check that all the requests fail. Otherwise this is going to take a *long* time to run. Will have a think...